server/src/modules/profile/util.service.ts

```diff
@@ -5,10 +5,28 @@ import { FilesRepository } from '@modules/files/repository';
 import { File } from '@entities/file.entity';
 import { IProfileUtilService } from '@modules/profile/interfaces/IService';
 import { CreateFileDto } from '@modules/files/dto/index';
+import { ProfileUpdateDto } from './dto';
+import { RequestContext } from '@modules/request-context/service';
+import { AUDIT_LOGS_REQUEST_CONTEXT_KEY } from '@modules/app/constants';
+import { User } from '@entities/user.entity';
 
 @Injectable()
 export class ProfileUtilService implements IProfileUtilService {
-  constructor(protected readonly filesRepository: FilesRepository, protected userRepository: UserRepository) {}
+  constructor(protected readonly filesRepository: FilesRepository, protected readonly userRepository: UserRepository) {}
+
+  private async setAuditLogForUser(userId: string, manager: EntityManager): Promise<void> {
+    const user = await manager.findOneOrFail(User, {
```

Comment:
You can pass user dto from controller to service functions. if it it applies for other events, please follow that instead of querying it one more time

Reply:
`manager.update`  only returns the affected rows and not the updated user so i have to query the user regardless because the dto does not contain sufficient data for audit logs